app/database.py

```diff
@@ -57,4 +57,29 @@ def createNewChallenge(creator_id, title, path, dt_submit, dt_finish):
         dt_finish=dt_finish,
         finished=False)
     db.session.add(newChallenge)
-    db.session.commit()
\ No newline at end of file
+    db.session.commit()
+
+# Given a Challenge model, find the submission with the fewest articles in its path (the winner of the challenge)
+def findWinner(challenge):
+    maxArticles = 51 # For what we currently have, each path has a maximum length of 50 articles
+    winningSubmission = None
+    for submission in challenge.attempts:
+        # If there's a tie, the earliest submission should be the winner, and in the database, it automatically sorts via attempt submission
+        if submission.article_no < maxArticles:
+            maxArticles = submission.article_no
+            winningSubmission = submission.id
+    return winningSubmission
+
+# Query the database to see if a challenge is finished. If so, update the database to store who won the challenge.
+def updateDB():
```

Comment:
Minor nitpick bu i think a different name for the function would be better? updateDB implies a lot of things, while you are just checking if challenges are finished?

Reply:
While making the code I couldn't think of a better function name. Maybe something like `checkChallenges()` or `checkChallengesCompleted()` ?

Reply:
checkChallengesCompleted() is better imo, if a bit wordy